ArduCopter/mode.cpp

```diff
@@ -368,10 +368,13 @@ bool Copter::set_mode(Mode::Number mode, ModeReason reason)
 #endif
 
 #if AP_FENCE_ENABLED
-    // pilot requested flight mode change during a fence breach indicates pilot is attempting to manually recover
-    // this flight mode change could be automatic (i.e. fence, battery, GPS or GCS failsafe)
-    // but it should be harmless to disable the fence temporarily in these situations as well
-    fence.manual_recovery_start();
+    // scripted changes are by definition not manual
+    if (reason != ModeReason::SCRIPTING) {
```

Comment:
If fence action is zero we really ought to do nothing - including this manual recovery work.

Reply:
Fixed